build_tools/ci/cpu_comparison/run.py

```diff
@@ -37,35 +38,48 @@ def run_conv_test(config, filename, n_repeats):
     return True
 
 
-class BaseTemplate(ABC):
+class BaseTest(ABC):
     """
-    Base class to be inherited by any new dispatch/compute op template.
-    The derived instances would therefore be created specifying the intended target
-    device(s) they're to be run; and would accordingly be `run` if the intended
-    target device(s) contains the `target_device` supplied via command-line.
+    Base class to be inherited by all tests. The derived instances will be
+    created specifying the intended target device(s) they're to be run on; and
+    will accordingly be `run` if the intended target device(s) contains
+    the `target_device` found in `config`. The default set of targets to run on
+    is the singleton set `["npu1_4col"]`.
+
+    An instance of this class has a member `aie_compilation_flags`
+    which are additional flags to be passed to the AIE backend compiler.
+
+    Compilation flags can therefore be injected in 2 ways:
+    1) via the constructor of this base class
+    2) via the `add_aie_compilation_flags` method
     """
 
-    def __init__(self, run_on_target, additional_aie_compilation_flags=None):
+    def __init__(self, run_on_target=["npu1_4col"], aie_compilation_flags=[]):
```

Comment:
Did you specifically change this on purpose from `None` to `[]`, and if so, why? In general, I try to avoid this because it can lead to quirky bahviour/issues: https://stackoverflow.com/questions/43255481/python-default-list-in-function

I saw your comment on the other PR (see below), but I am not sure what that has to do with this? Every object has it's own `self.aie_compilation_flags`?

> Just noticed: you are overriding the additional flags for every test which runs subsequent to the one that appends. I'll try and think of a way to add flags to a test in isolation

Reply:
>  config.add_additional_aie_compilation_flag(flag)

After this, the config has more flags for all subsequent tests. It's the appending to config from a specific test which looks bad. The config should only contain things which are common across all tests.

Reply:
> Did you specifically change this on purpose from None to [], and if so, why?

It was intentional to be consistent with run_on_target, but I didn't realize there were potential issues. I'll change it back to None